src/OpenTK.Graphics/OpenGL/GL.Manual.cs

```diff
@@ -5,34 +5,58 @@ namespace OpenTK.Graphics.OpenGL
 {
     // FIXME: Remove this when it's fixed
     // This is here because there in the gl.xml
-    // one of the paramteres for "glSampleMaskIndexedNV"
+    // one of the parameters for "glSampleMaskIndexedNV"
     // is marked with a group named this, but this group is never referenced
     // anywhere else in the file.
     public enum SampleMaskNV
     { }
 
     public static unsafe partial class GL
     {
-        public static void ShaderSource(uint shader, string str)
+        public static void ShaderSource(uint shader, string shaderText)
         {
-            IntPtr str_iptr = Marshal.StringToCoTaskMemAnsi(str);
-            int length = str.Length;
-            GL.ShaderSource(shader, 1, (byte**)&str_iptr, length);
-            Marshal.FreeCoTaskMem(str_iptr);
+            if (string.IsNullOrEmpty(shaderText))
+            {
+                throw new ArgumentNullException(nameof(shaderText));
+            }
```

Comment:
I think we should handle this differently. There might be some weird reason why someone might need to pass a null string into this function.
This is not an explicit error case of `glShaderSource`.

Reply:
Until somebody can explain to my why passing a null makes sense, it should be validated as it is, imho.

Reply:
OpenGL is such a complicated API that you can't really know all of the valid ways to use the API.

Generally we have opted for making convenience overloads without doing much error checking to allow for usages that where not known at the time of writing the code. It's a way of future proofing the API.

Reply:
Alright, from a professional point of view I disagree, however, I will comply :)